specifyweb/frontend/js_src/lib/components/QueryBuilder/ResultsTable.tsx

```diff
@@ -120,6 +118,13 @@ function Row({
   );
   const viewUrl = typeof resource === 'object' ? resource.viewUrl() : undefined;
 
+  const splitIds: RA<number> | undefined =
+    typeof result[0] === 'string' && result[0].includes(',')
+      ? result[0].split(',').map(Number)
+      : undefined;
+
+  const [isIdListOpen, _, __, handleToggle] = useBooleanState(false);
```

Comment:
(optional) while I normally advise for using useBooleanState, given that we don't yet have virtual rendering, the `Row` component may be called many times, and so needs to be as efficient as possible.
given that, you might want to use a regular useState here as it would have slightly less overhead than useBooleanState (because useBooleanState calls several other hooks inside of it, including useCallback 3 times)

Reply:
Nice idea! Do you think it would be better to have just one state at the top? Then, we can pass a callback (as a prop ig) to row which calls it to set the list of ids? Basically, shift the state responsible for displaying ids at the top and pass a callback to set the state to children (rows). I think that should be more efficient?

Reply:
> Nice idea! Do you think it would be better to have just one state at the top? Then, we can pass a callback (as a prop ig) to row which calls it to set the list of ids? Basically, shift the state responsible for displaying ids at the top and pass a callback to set the state to children (rows). We would need to move dialog ofc to the parent too. I think that should be more efficient?

yes, that would be even better! that's what we do in the attachments gallery if I remember right